packages/core/src/main/notification/NotificationService.ts

```diff
@@ -231,8 +240,55 @@ export class NotificationService extends EventEmitter {
     source: PayloadBundleSource,
     dryRun: boolean = false,
   ): Promise<HandledEventPayload> {
+    const coreCryptoClient = this.coreCryptoClientProvider();
     switch (event.type) {
-      // Encrypted events
+      case Events.CONVERSATION_EVENT.MLS_WELCOME_MESSAGE:
+        if (!coreCryptoClient) {
+          // TODO throw proper error
+          throw new Error('TODO');
+        }
+        const data = Decoder.fromBase64(event.data).asBytes;
+        // We extract the groupId from the welcome message and let coreCrypto store this group
+        const newGroupId = await coreCryptoClient.processWelcomeMessage(data);
+        const saveGroupId = (conversationId: QualifiedId, groupId: Uint8Array): void => {
+          // 🛑 TODO save the mapping between conversationId > groupID (not localStorage)
+          localStorage.setItem(`${conversationId.id}@${conversationId.domain}`, Encoder.toBase64(groupId).asString);
+        };
+        saveGroupId(event.qualified_conversation ?? {id: event.conversation, domain: ''}, newGroupId);
+        // The groupId can then be sent back to the consumer
+        return {
+          event,
+          mappedEvent: ConversationMapper.mapConversationEvent(
+            {...event, data: Encoder.toBase64(newGroupId).asString},
+            source,
+          ),
+        };
+
+      case Events.CONVERSATION_EVENT.MLS_MESSAGE_ADD:
+        if (!coreCryptoClient) {
+          // TODO throw proper error
+          throw new Error('TODO');
+        }
+        const encryptedData = Decoder.fromBase64(event.data).asBytes;
+        const findGroupId = (conversationId: QualifiedId) => {
+          // TODO fetch the groupId in DB from the conversatioId
+          const base64groupId = localStorage.getItem(`${conversationId.id}@${conversationId.domain}`);
```

Comment:
We need to get the groupId from the indexeddb store we used when the welcome message arrived